sdk/src/main/java/com/akkaserverless/javasdk/AkkaServerless.java

```diff
@@ -495,6 +496,30 @@ public AkkaServerless registerValueEntity(
     return this;
   }
 
+  /**
+   * Register a value based entity using a ValueEntityProvider.
+   *
+   * @return This stateful service builder.
+   */
+  public <T> AkkaServerless register(ValueEntityProvider provider) {
+    return register(provider, ValueEntityOptions.defaults());
+  }
+
+  /**
+   * Register a value based entity using a ValueEntityProvider with user defined ValueEntityOptions.
+   *
+   * @return This stateful service builder.
+   */
+  public <T> AkkaServerless register(ValueEntityProvider provider, ValueEntityOptions options) {
```

Comment:
An alternative would be that we make ValueEntityProvider a builder, so that options can be specified via the ValueEntityProvider. 

```
CustomerValueEntityProvider.of(context -> new CustomerValueEntity()).withOptions(...)
```

The advantage is that it's easier for us to evolve the builder with more optional properties, than having overloaded register methods.